python-package/xgboost/sklearn.py

```diff
@@ -221,13 +223,17 @@ def set_params(self, **params):
 
         return self
 
-    def get_params(self, deep=False):
+    def get_params(self, deep=True):
+        # pylint: disable=attribute-defined-outside-init
         """Get parameters."""
-        params = super(XGBModel, self).get_params(deep=deep)
+        #Based on: https://stackoverflow.com/questions/59248211
+        params = super().get_params(deep)
+        cp = copy.copy(self)
```

Comment:
Copying involves a serialization and unserialization of model, performance is one thing, another drawback is all caches of `DMatrix` are removed during copy.  Will look into the technique further.

Reply:
I was thinking that `DMatrix` is involved only during `fit`, so `__init__` is light-weight but for sure it need to be checked.